services/debian/debian.service.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const Joi = require('joi')
+const { latest, renderVersionBadge } = require('../version')
+const { BaseJsonService, NotFound } = require('..')
+
+const schema = Joi.array()
+  .items(
+    Joi.object().pattern(
+      /./,
+      Joi.object()
+        .pattern(/./, Joi.object().pattern(/./, Joi.object()))
+        .required()
+    ) // Optional, missing means not found
+  )
+  .required()
+
+const defaultDistribution = 'stable'
+
+module.exports = class Debian extends BaseJsonService {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'debian/v',
+      pattern: ':distribution?/:packageName',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Debian package',
+        namedParams: { distribution: 'unstable', packageName: 'apt' },
+        staticPreview: renderVersionBadge({ version: '1.8.0' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'debian' }
+  }
+
+  async handle({ distribution = defaultDistribution, packageName }) {
+    const data = await this._requestJson({
+      schema,
+      url: `https://api.ftp-master.debian.org/madison?f=json&s=${distribution}&package=${packageName}`,
+    })
+    if (!data.length) {
+      throw new NotFound()
+    }
+    // Distribution can change compared to request, for example requesting
+    // "stretch" may map to "stable" in results, so can't use value from
+    // request as is. Just take the first one instead.
+    for (const resultdist of Object.keys(data[0][packageName])) {
```

Comment:
I'm not familiar with this API. Is there any valid case where `data.length` can be >1?

Reply:
Probably not. Would you like a case where it is handled somehow differently?

Reply:
If we're getting back arrays which should only contain zero or one elements (and we don't know what do do if we get more), can we validate that in the schema.

I can't remember off the top of my head, but I think chaining `.single()` without `required()` will pass zero or one elements. If not, chaining `.min(0).max(1)` will do the job.

Reply:
I think `.single()` is the other way around (get single item as if it was an array). `.max(1).required()` seems to do the right thing.